src/Microsoft.Health.Fhir.Api/Features/ActionResults/ResourceActionResult.cs

```diff
@@ -68,6 +68,11 @@ public override Task ExecuteResultAsync(ActionContext context)
 
             foreach (KeyValuePair<string, StringValues> header in Headers)
             {
+                if (response.Headers.ContainsKey(header.Key))
```

Comment:
Could we add a unit test to show it properly handles duplicate headers now?

Reply:
Done. Please review